File: be/src/main/java/plus/easydo/dnf/controller/GameRoleController.java
package plus.easydo.dnf.controller;

import cn.dev33.satoken.annotation.SaCheckLogin;
import cn.dev33.satoken.annotation.SaCheckPermission;
import cn.dev33.satoken.stp.StpUtil;
import io.swagger.v3.oas.annotations.Operation;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import plus.easydo.dnf.dto.AccountCargoDto;
import plus.easydo.dnf.dto.EditGameRoleItemDto;
import plus.easydo.dnf.dto.OtherDataDto;
import plus.easydo.dnf.entity.AccountCargo;
import plus.easydo.dnf.entity.CharacInfo;
import plus.easydo.dnf.entity.UserItems;
import plus.easydo.dnf.qo.CharacInfoQo;
import plus.easydo.dnf.service.GamePostalService;
import plus.easydo.dnf.service.GameRoleService;
import plus.easydo.dnf.vo.DataResult;
import plus.easydo.dnf.vo.RoleItemVo;
import plus.easydo.dnf.vo.OtherDataVo;
import plus.easydo.dnf.vo.R;

import java.util.List;

/**
 * @author laoyu
 * @version 1.0
 * @description 游戏角色相关
 * @date 2023/10/14
 */
@RestController
@RequestMapping("/api/gameRole")
@RequiredArgsConstructor
public class GameRoleController {

    private final GameRoleService gameRoleService;

    private final GamePostalService gamePostalService;

    @Operation(summary = "所有角色列表")
    @SaCheckLogin
    @GetMapping("/list")
    public R<List<CharacInfo>> roleList(@RequestParam(value = "name", required = false)String name){

        return DataResult.ok(gameRoleService.roleList(StpUtil.getLoginIdAsLong(),name));
    }

    @Operation(summary = "所有角色列表")
    @SaCheckLogin
    @GetMapping("/allList")
    public R<List<CharacInfo>> allList(@RequestParam(value = "name", required = false)String name){
        return DataResult.ok(gameRoleService.roleList(null,name));
    }

    @Operation(summary = "分页查询")
    @SaCheckPermission("gameRole")
    @PostMapping("/page")
    public R<List<CharacInfo>> gameRolePage(@RequestBody CharacInfoQo characInfoQo){
        return DataResult.ok(gameRoleService.page(characInfoQo));
    }


    @Operation(summary = "详情")
    @SaCheckPermission("gameRole")
    @GetMapping("/info/{characNo}")
    public R<CharacInfo> gameRoleInfo(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gameRoleService.info(characNo));
    }

    @Operation(summary = "更新角色信息")
    @SaCheckPermission("gameRole.update")
    @PostMapping("/update")
    public R<Boolean> gameRoleUpdate(@RequestBody CharacInfo characInfo){
        return DataResult.ok(gameRoleService.update(characInfo));
    }

    @Operation(summary = "删除角色")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/remove/{characNo}")
    public R<Boolean> gameRoleRemove(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gameRoleService.remove(characNo));
    }

    @Operation(summary = "恢复角色")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/recover/{characNo}")
    public R<Boolean> gameRoleRecover(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gameRoleService.recover(characNo));
    }


    @Operation(summary = "获取角色物品")
    @SaCheckPermission("gameRole")
    @GetMapping("/getRoleItem/{characNo}/{type}")
    public R<RoleItemVo> getRoleItem(@PathVariable("characNo") Long characNo, @PathVariable("type") String type){
        return DataResult.ok(gameRoleService.getRoleItem(characNo,type));
    }

    @Operation(summary = "更新角色物品")
    @SaCheckPermission("gameRole.update")
    @PostMapping("/updateRoleItem")
    public R<Boolean> updateRoleItem(@RequestBody EditGameRoleItemDto editGameRoleItemDto){
        return DataResult.ok(gameRoleService.updateRoleItem(editGameRoleItemDto));
    }

    @Operation(summary = "清空角色物品")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/cleanRoleItem/{characNo}/{type}")
    public R<Boolean> cleanRoleItem(@PathVariable("characNo") Long characNo,@PathVariable("type") Long type){
        return DataResult.ok(gameRoleService.cleanItem(characNo,type));
    }

    @Operation(summary = "开启角色左右槽")
    @SaCheckPermission("gameRole.openLeftAndRight")
    @GetMapping("/openLeftAndRight/{characNo}")
    public R<Boolean> openLeftAndRight(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gameRoleService.openLeftAndRight(characNo));
    }

    @Operation(summary = "清空角色邮件")
    @SaCheckPermission("gameRole.update")
    @PostMapping("/cleanMail/{characNo}")
    public R<Boolean> cleanMail(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gamePostalService.cleanCharacMail(characNo));
    }

    @Operation(summary = "获取角色其他信息")
    @SaCheckPermission("gameRole")
    @GetMapping("/getOtherData/{characNo}")
    public R<OtherDataVo> getOtherData(@PathVariable("characNo") Long characNo){
        return DataResult.ok(gameRoleService.getOtherData(characNo));
    }

    @Operation(summary = "修改角色其他信息")
    @SaCheckPermission("gameRole.update")
    @PostMapping("/setOtherData")
    public R<Boolean> setOtherData(@RequestBody OtherDataDto otherDataDto){
        return DataResult.ok(gameRoleService.setOtherData(otherDataDto));
    }

    @Operation(summary = "获取时装信息")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/items/{characNo}")
    public R<List<UserItems>> roleItems(@PathVariable("characNo") Long characNo) {
        return DataResult.ok(gameRoleService.roleItems(characNo));
    }

    @Operation(summary = "删除单件时装")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/removeItems/{uiId}")
    public R<Boolean> removeItems(@PathVariable("uiId") Long uiId) {
        return DataResult.ok(gameRoleService.removeItems(uiId));
    }

    @Operation(summary = "清空所有时装")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/cleanItems/{characNo}")
    public R<Boolean> cleanItems(@PathVariable("characNo") Long characNo) {
        return DataResult.ok(gameRoleService.cleanItems(characNo));
    }

    @Operation(summary = "获取账号金库信息")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/getAccountCargo/{characNo}")
    public R<AccountCargo> getAccountCargo(@PathVariable("characNo") Long characNo) {
        return DataResult.ok(gameRoleService.getAccountCargo(characNo));
    }

    @Operation(summary = "开通账号金库")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/createAccountCargo/{characNo}")
    public R<Boolean> createAccountCargo(@PathVariable("characNo") Long characNo) {
        return DataResult.ok(gameRoleService.createAccountCargo(characNo));
    }

    @Operation(summary = "删除账号金库")
    @SaCheckPermission("gameRole.update")
    @GetMapping("/removeAccountCargo/{characNo}")
    public R<Boolean> removeAccountCargo(@PathVariable("characNo") Long characNo) {
        return DataResult.ok(gameRoleService.removeAccountCargo(characNo));
    }

    @Operation(summary = "修改账号金库信息")
    @SaCheckPermission("gameRole.update")
    @PostMapping("/updateAccountCargo")
    public R<Boolean> updateAccountCargo(@RequestBody AccountCargoDto accountCargoDto) {
        return DataResult.ok(gameRoleService.updateAccountCargo(accountCargoDto));
    }

}

File: antd/src/services/dnf-admin/gameRoleController.ts
// @ts-ignore
/* eslint-disable */
import { request } from 'umi';

/** 所有角色列表 GET /api/gameRole/allList */
export async function allList(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.allListParams,
  options?: { [key: string]: any },
) {
  return request<API.RListCharacInfo>('/api/gameRole/allList', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 清空所有时装 GET /api/gameRole/cleanItems/${param0} */
export async function cleanItems(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.cleanItemsParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/cleanItems/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 清空角色邮件 POST /api/gameRole/cleanMail/${param0} */
export async function cleanMail(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.cleanMailParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/cleanMail/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 清空角色物品 GET /api/gameRole/cleanRoleItem/${param0}/${param1} */
export async function cleanRoleItem(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.cleanRoleItemParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, type: param1, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/cleanRoleItem/${param0}/${param1}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 开通账号金库 GET /api/gameRole/createAccountCargo/${param0} */
export async function createAccountCargo(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.createAccountCargoParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/createAccountCargo/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 获取账号金库信息 GET /api/gameRole/getAccountCargo/${param0} */
export async function getAccountCargo(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getAccountCargoParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RAccountCargo>(`/api/gameRole/getAccountCargo/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 获取角色其他信息 GET /api/gameRole/getOtherData/${param0} */
export async function getOtherData(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getOtherDataParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.ROtherDataVo>(`/api/gameRole/getOtherData/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 获取角色物品 GET /api/gameRole/getRoleItem/${param0}/${param1} */
export async function getRoleItem(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getRoleItemParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, type: param1, ...queryParams } = params;
  return request<API.RRoleItemVo>(`/api/gameRole/getRoleItem/${param0}/${param1}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 详情 GET /api/gameRole/info/${param0} */
export async function gameRoleInfo(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.gameRoleInfoParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RCharacInfo>(`/api/gameRole/info/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 获取时装信息 GET /api/gameRole/items/${param0} */
export async function roleItems(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.roleItemsParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RListUserItems>(`/api/gameRole/items/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 所有角色列表 GET /api/gameRole/list */
export async function roleList(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.roleListParams,
  options?: { [key: string]: any },
) {
  return request<API.RListCharacInfo>('/api/gameRole/list', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 开启角色左右槽 GET /api/gameRole/openLeftAndRight/${param0} */
export async function openLeftAndRight(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.openLeftAndRightParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/openLeftAndRight/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 分页查询 POST /api/gameRole/page */
export async function gameRolePage(body: API.CharacInfoQo, options?: { [key: string]: any }) {
  return request<API.RListCharacInfo>('/api/gameRole/page', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 恢复角色 GET /api/gameRole/recover/${param0} */
export async function gameRoleRecover(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.gameRoleRecoverParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/recover/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 删除角色 GET /api/gameRole/remove/${param0} */
export async function gameRoleRemove(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.gameRoleRemoveParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/remove/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 删除账号金库 GET /api/gameRole/removeAccountCargo/${param0} */
export async function removeAccountCargo(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.removeAccountCargoParams,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/removeAccountCargo/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 删除单件时装 GET /api/gameRole/removeItems/${param0} */
export async function removeItems(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.removeItemsParams,
  options?: { [key: string]: any },
) {
  const { uiId: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/gameRole/removeItems/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 修改角色其他信息 POST /api/gameRole/setOtherData */
export async function setOtherData(body: API.OtherDataDto, options?: { [key: string]: any }) {
  return request<API.RBoolean>('/api/gameRole/setOtherData', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 更新角色信息 POST /api/gameRole/update */
export async function gameRoleUpdate(body: API.CharacInfo, options?: { [key: string]: any }) {
  return request<API.RBoolean>('/api/gameRole/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 修改账号金库信息 POST /api/gameRole/updateAccountCargo */
export async function updateAccountCargo(
  body: API.AccountCargoDto,
  options?: { [key: string]: any },
) {
  return request<API.RBoolean>('/api/gameRole/updateAccountCargo', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 更新角色物品 POST /api/gameRole/updateRoleItem */
export async function updateRoleItem(
  body: API.EditGameRoleItemDto,
  options?: { [key: string]: any },
) {
  return request<API.RBoolean>('/api/gameRole/updateRoleItem', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}
